src/storage/src/store_impl.rs

```diff
@@ -643,27 +645,32 @@ impl StateStoreImpl {
                 .with_weighter(|_: &HummockSstableObjectId, value: &Box<Sstable>| {
                     u64::BITS as usize / 8 + value.estimate_size()
                 })
-                .storage();
+                .storage(Engine::Large);
 
             if !opts.meta_file_cache_dir.is_empty() {
                 builder = builder
-                    .with_device_config(
-                        DirectFsDeviceOptionsBuilder::new(&opts.meta_file_cache_dir)
+                    .with_device_options(
+                        DirectFsDeviceOptions::new(&opts.meta_file_cache_dir)
                             .with_capacity(opts.meta_file_cache_capacity_mb * MB)
-                            .with_file_size(opts.meta_file_cache_file_capacity_mb * MB)
-                            .build(),
-                    )
-                    .with_indexer_shards(opts.meta_file_cache_indexer_shards)
-                    .with_flushers(opts.meta_file_cache_flushers)
-                    .with_reclaimers(opts.meta_file_cache_reclaimers)
-                    .with_buffer_pool_size(opts.meta_file_cache_flush_buffer_threshold_mb * MB) // 128 MiB
-                    .with_clean_region_threshold(
-                        opts.meta_file_cache_reclaimers + opts.meta_file_cache_reclaimers / 2,
+                            .with_file_size(opts.meta_file_cache_file_capacity_mb * MB),
                     )
                     .with_recover_mode(opts.meta_file_cache_recover_mode)
-                    .with_recover_concurrency(opts.meta_file_cache_recover_concurrency)
                     .with_compression(opts.meta_file_cache_compression)
-                    .with_runtime_config(opts.meta_file_cache_runtime_config.clone());
+                    .with_runtime_options(opts.meta_file_cache_runtime_config.clone())
+                    .with_large_object_disk_cache_options(
+                        LargeEngineOptions::new()
+                            .with_indexer_shards(opts.meta_file_cache_indexer_shards)
+                            .with_flushers(opts.meta_file_cache_flushers)
+                            .with_reclaimers(opts.meta_file_cache_reclaimers)
+                            .with_buffer_pool_size(
+                                opts.meta_file_cache_flush_buffer_threshold_mb * MB,
+                            ) // 128 MiB
+                            .with_clean_region_threshold(
```

Comment:
just a question: Is the value of `clean_region_threshold` 1.5 times the value of `reclaimers`? Is this a specially tuned parameter for RW environments?

Reply:
Yes, it is x1.5. But I think it may not affect that much.